element-connector/src/main/java/org/eclipse/californium/elements/util/SslContextUtil.java

```diff
@@ -524,6 +740,40 @@ public static SSLContext createSSLContext(String alias, PrivateKey privateKey, X
 	}
 
 	/**
+	 * Scheme specific input stream factory.
+	 * 
+	 * @see SslContextUtil#configure(String, InputStreamFactory)
+	 */
+	public static interface InputStreamFactory {
+
+		/**
+		 * Create input stream of the provided URI.
+		 * 
+		 * @param uri URI to read
+		 * @return input stream
+		 * @throws IOException if creating the input stream fails
+		 */
+		InputStream create(String uri) throws IOException;
+	}
+
+	/**
+	 * Input stream factory for classpath resources.
+	 */
+	private static class ClassLoaderInputStreamFactory implements InputStreamFactory {
+
+		@Override
+		public InputStream create(String uri) throws IOException {
+			String resource = uri.substring(CLASSPATH_SCHEME.length());
+			InputStream inStream = SslContextUtil.class.getClassLoader().getResourceAsStream(resource);
```

Comment:
any reason to use :
` SslContextUtil.class.getClassLoader().getResourceAsStream(resource)`
instead of 
` SslContextUtil.class.getResourceAsStream(resource)`
?

Reply:
What should be the advantage of that? Do you expect some certificate store located in the class package? If not, the class loader is the better choice.

Reply:
It seems just lighter but you right the behavior is not exactly the same.

Any reason to use the class SslContextUtil class loader ? It is better than  `ClassLoader.getSystemResourceAsStream()`

Reply:
Mainly, because it reuses the working DtlsTestTools.loadKeyStore().
If you feel, that the system classloader is a better choice. feel free to change AND test it (in the different environments e.g. OSGi and Android).

Reply:
By the way this PU doesn't change the class loader. It changes the possible keystore type from fixed to configurable and the reading to an extendable configurable solution to enable android to plugin a AssetManger based solution.

Reply:
Just to add: If you prefer to use the system classloader, you may just configure your own InputStreamFactory, which uses that instead. I prefer that classloader I used.

Reply:
That was just curiosity, personally I didn't know when to use one or another and I just asked in case you know.